### ISSUE TITLE
Difficulty with the order of certain small fp groups

### ISSUE THREAD

**Issue by grahamknockillaree** [gap-system/gap#3310]
### Observed behaviour
```
gap> C:=CyclicGroup(4);;
gap> G:=CyclicGroup(4);; 
gap> f:=GeneratorsOfGroup(G);;
gap> H:=Group([f[1]^2]);;
gap> map:=GroupHomomorphismByFunction(H,G,x->x);;
gap> RH:=ResolutionFiniteGroup(H,5);;
gap> RG:=ResolutionFiniteGroup(G,5);;
gap> eq:=EquivariantChainMap(RH,RG,map);;
gap> T:=HomToIntegersModP(eq,2);;
gap> Coh:=Cohomology(T,2);
[ f1 ] -> [ f1 ]
gap> A:=Source(Coh);
<fp group on the generators [ f1 ]>
gap> IsFpGroup(A);
true
gap> NamesOfComponents(A);
[ "GeneratorsOfMagmaWithInverses", "FreeGeneratorsOfFpGroup", 
  "FreeGroupOfFpGroup" ]
gap> RelatorsOfFpGroup(A);
[ f1^2, <identity ...> ]
gap> Order(A);
Error, List Element: <list>[1] must have an assigned value in
  while not r in ccr[offset + r[1]] do
    AddSet( ccr[offset + r[1]], Immutable( r ) );
    r := Concatenation( r{[ 2 .. Length( r ) ]}, r{[ 1 ]} );
od; at /usr/share/gap-4.10.0/lib/sgpres.gi:2936 called from 
NEWTC_DoCosetEnum( freegens, freerels, subgens, aug, trace 
 ) at /usr/share/gap-4.10.0/lib/sgpres.gi:3200 called from
NEWTC_CosetEnumerator( FreeGeneratorsOfFpGroup( G ), RelatorsOfFpGroup( G ), 
 GeneratorsOfGroup( H ), true, false 
 ) at /usr/share/gap-4.10.0/lib/grpfp.gi:3856 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:75
you can 'return;' after assigning a value
brk> 
```

### Expected behaviour
```
> gap> G:=CyclicGroup(4);;
> gap> f:=GeneratorsOfGroup(G);;
> gap> H:=Group([f[1]^2]);;
> gap> map:=GroupHomomorphismByFunction(H,G,x->x);;
> gap> RH:=ResolutionFiniteGroup(H,5);;
> gap> RG:=ResolutionFiniteGroup(G,5);;
> gap> eq:=EquivariantChainMap(RH,RG,map);;
> gap> T:=HomToIntegersModP(eq,2);;
> gap> Coh:=Cohomology(T,2);
> [ f1 ] -> [ f1 ]
> gap> A:=Coh!.Source;
> <fp group on the generators [ f1 ]>
> gap> Order(A);
> 2
```

### Copy and paste GAP banner (to tell us about your setup)
```
 ┌───────┐   GAP 4.10.0 of 01-Nov-2018
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-default64
 Configuration:  gmp 6.1.2
 Loading the library and packages ...
 Packages:   AClib 1.3.1, Alnuth 3.1.0, AtlasRep 1.5.1, AutoDoc 2018.09.20, 
             AutPGrp 1.10, CRISP 1.4.4, Cryst 4.1.18, CrystCat 1.1.8, 
             CTblLib 1.2.2, FactInt 1.6.2, FGA 1.4.0, GAPDoc 1.6.2, 
             IRREDSOL 1.4, LAGUNA 3.9.0, Polenta 1.3.8, Polycyclic 2.14, 
             PrimGrp 3.3.2, RadiRoot 2.8, ResClasses 4.7.1, SmallGrp 1.3, 
             Sophus 1.24, SpinSym 1.5, TomLib 1.2.7, TransGrp 2.0.4, 
             utils 0.59
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
```

**Comment by markuspf** [comment]
Easier to reproduce:

```
gap> F := FreeGroup(1);;
gap> rels := [F.1^2, One(F)];;
gap> G := F / rels;;
gap> Order(G);
```

the cause is the identity element in the relators, which I think we can just remove in a cleanup step, unless @hulpke has any objections.

**Comment by grahamknockillaree** [comment]
That would be perfect.

And note the following behaviour which suggests your solution is already partially implemented in GAP v4.10.0.

gap> F := FreeGroup(1);;
gap> rels := [F.1^2, One(F)];;
gap> G := F / rels;;
gap> Order(G);
Error, List Element: <list>[1] must have an assigned value in
  while not r in ccr[offset + r[1]] do
    AddSet( ccr[offset + r[1]], Immutable( r ) );
    r := Concatenation( r{[ 2 .. Length( r ) ]}, r{[ 1 ]} );
od; at /usr/share/gap-4.10.0/lib/sgpres.gi:2936 called from 
NEWTC_DoCosetEnum( freegens, freerels, subgens, aug, trace 
 ) at /usr/share/gap-4.10.0/lib/sgpres.gi:3200 called from
NEWTC_CosetEnumerator( FreeGeneratorsOfFpGroup( G ), RelatorsOfFpGroup( G ), 
 GeneratorsOfGroup( H ), true, false 
 ) at /usr/share/gap-4.10.0/lib/grpfp.gi:3856 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:4
you can 'return;' after assigning a value
brk> quit;
#I  Options stack has been reset
gap> Order(G);
2

Thanks,

Graham


School of Mathematics, Statistics & Applied Mathematics
National University of Ireland, Galway
http://hamilton.nuigalway.ie
tel: 091 493011
________________________________________

**Comment by markuspf** [comment]
Note that you probably shouldn't trust any results GAP gives you after you entered a break loop and quit it again...

**Comment by fingolfin** [comment]
Fixed by PR #3311

**Comment by grahamknockillaree** [comment]
Thanks!

And I agree that it is good to continue to allow identity relators in presentations of fp groups. The identity relator(s) is important when considering the homotopy type of the CW complex associated to the presentation.

Thanks again for your quick response.

Graham


School of Mathematics, Statistics & Applied Mathematics
National University of Ireland, Galway
http://hamilton.nuigalway.ie
tel: 091 493011
________________________________________
